src/jsonrpc/miner_jsonrpc_txn.erl

```diff
@@ -96,3 +99,14 @@ parse_location(#{ <<"lat">> := LatIn,
             {error, {invalid_location, {LatIn, LonIn}}}
     end;
 parse_location(_Other) -> {error, no_valid_location_found}.
+
+-spec parse_integer(binary() | undefined) -> integer() | undefined.
+parse_integer(undefined) ->
+    undefined;
+parse_integer(Value) when is_binary(Value) ->
+    try 
+        binary_to_integer(Value)
+    catch
+        _:_ ->
+            {error, {invalid_integer, Value}}
```

Comment:
this doesn't return `undefined` like in the spec.

Reply:
gah, ok lemme put up another PR.. you may want to leave review comments before approving 😛